clang/lib/Driver/ToolChains/ClassicFlang.cpp

```diff
@@ -1006,6 +1006,28 @@ void ClassicFlang::ConstructJob(Compilation &C, const JobAction &JA,
   LowerCmdArgs.push_back("-x"); LowerCmdArgs.push_back("189"); LowerCmdArgs.push_back("0x10");
   LowerCmdArgs.push_back("-y"); LowerCmdArgs.push_back("189"); LowerCmdArgs.push_back("0x4000000");
 
+  bool AssociativeMath = false;
+  bool SignedZeros = true;
+  for (const Arg *A : Args) {
+    switch (A->getOption().getID()) {
+    case options::OPT_fassociative_math:    AssociativeMath = true;   break;
+    case options::OPT_fno_associative_math: AssociativeMath = false;  break;
+    case options::OPT_fsigned_zeros:        SignedZeros = true;       break;
+    case options::OPT_fno_signed_zeros:     SignedZeros = false;      break;
+    default: break;
+    }
+  }
+  if (!SignedZeros && !NeedIEEE) {
+    LowerCmdArgs.push_back("-x");
+    LowerCmdArgs.push_back("216");
+    LowerCmdArgs.push_back("0x8");
+  }
+  if (AssociativeMath && !SignedZeros && !NeedIEEE) {
```

Comment:
This implementation allows an earlier option to suppress the effect of a subsequent option, which contradicts the usual understanding of how compiler options are processed. I would prefer that Classic Flang processes options the same way as Clang would; if it doesn't, then I think we should clearly document the difference in behaviour. @shivaramaarao What do you think?